src/neoxp/Node/NodeUtility.cs

```diff
@@ -138,5 +142,127 @@ public static void SignOracleResponseTransaction(ProtocolSettings settings, Expr
 
             return tx;
         }
+
+        public const byte Prefix_Contract = 8;
+        private const byte Prefix_NextAvailableId = 15;
+
+        private static int LastUsedContractId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.TryGet(key);
+            return (int)(BigInteger)item - 1;
+        }
+
+        private static void SetNextAvailableContractId(DataCache snapshot, int newId)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            item.Set(newId);
+        }
+        
+        private static int GetNextAvailableId(DataCache snapshot)
+        {
+            StorageKey key = new KeyBuilder(NativeContract.ContractManagement.Id, Prefix_NextAvailableId);
+            StorageItem item = snapshot.GetAndChange(key);
+            int value = (int)(BigInteger)item;
+            item.Add(1);
+            return value;
+        }
+
+        public static async Task<(ContractState contractState, IReadOnlyList<(string key, string value)> storagePairs)> DownloadParamsAsync(
```

Comment:
Can we change the name of this? `DownloadParamsAsync` isn't really clear what params it's doing. Maybe `DownloadContractStateAsync` (yes, I know it's also downloading storage but I think just saying state is fine)